src/app/services/status.ts

```diff
@@ -0,0 +1,24 @@
+import { api } from './api';
+import { UserStatus } from '@/interfaces/userStatus.type';
+
+type StatusResponse = {
+    message: string;
+    totalUserStatus: number;
+    allUserStatus: UserStatus[];
+}
+
+export const statusApi = api.injectEndpoints({
+    endpoints: (build) => ({
+        getAllStatus: build.query<StatusResponse, void>({
+            query: () => '/users/status',
+            providesTags: ['Status']
+          }),
+          getIdleStatus: build.query<StatusResponse, string>({
+            query: (state) => `/users/status?state=${state}`,
```

Comment:
Here instead of taking state, you should be directly returning the URL with the state as status, Because the endpoint name implies what's trying to do here.

Also do check, what if this changed from `getIdleStatus` to `getStatus`?

Reply:
yeah for now i have written the same as it is before

Reply:
What's the update on this?

Reply:
chnaging it to getStatus

Reply:
changed to getStatus